src/test_s2c_srv.c

```diff
@@ -393,19 +396,25 @@ int test_s2c(Connection *ctl, tcp_stat_agent *agent, TestOptions *testOptions,
             log_println(0, "S2C test error: can't create child process.");
           }
         }
-        memset(tmpstr, 0, 256);
-        for (i = 0; i < 5; i++) {  // read nettrace file name into "tmpstr"
-          ret = read(mon_pipe[0], tmpstr, 128);
-          // socket interrupted, try reading again
-          if ((ret == -1) && (errno == EINTR))
-            continue;
-          break;
-        }
+        packet_trace_running = wait_for_readable_fd(mon_pipe[0]);
+        if (packet_trace_running) {
+          memset(tmpstr, 0, 256);
+          for (i = 0; i < 5; i++) {  // read nettrace file name into "tmpstr"
+            ret = read(mon_pipe[0], tmpstr, 128);
```

Comment:
since read() returns bytes read or negative, would be more efficient (and safer) to use that instead of strlen() below.